pkg/unidling/api/types.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import (
+	kextapi "k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
+	IdledAtAnnotation      = "idling.alpha.openshift.io/idled-at"
+	UnidleTargetAnnotation = "idling.alpha.openshift.io/unidle-targets"
+	NeedPodsReason         = "NeedPods"
+)
+
+// RecordedScaleReference is a SubresourceReference to a scale subresource that also
+// has the previous replica count recorded
+type RecordedScaleReference struct {
+	// TODO: update this to a CrossVersionObjectReference once we pull in a Kube version
+	// with an unversioned apis/autoscaling
+	kextapi.SubresourceReference `json:",inline"`
```

Comment:
This is a known bad type, we shouldn't propagate it.  It uses kind instead of resource, it stuffs group information into an APIversion field, it suggests that auto-negotiation can't or won't happen.  You should make a subresource reference that works, group (required), version (optional, support negotiation), resource (this isn't a kind), name, subresource.

Also, looks like `CrossVersionObjectReference` has managed to reproduce the same flaws.